### PR TITLE
Timelock bouncing

### PR DESCRIPTION
**Pull request by lichen19853**
- Add bouncing
- Add bending
- Animation curve is still to be tweaked, but states are good to go

### CODE REVIEW

File: ios/RCTMotionManager.m [hunk 1]
```diff
@@ -68,42 +69,70 @@ - (BOOL) isFlatWithGravity:(CMAcceleration) gravity {
...
+    if (strongSelf->_lockState == RCTMotionManagerStateUnlocking
+        || strongSelf->_lockState == RCTMotionManagerStateBouncing) {
+      // An animation is going on, don't use sensor's input
       return;
     }
     if (motion == nil) { return; }
 
     CMAcceleration gravity = motion.gravity;
     if ([strongSelf isFlatWithGravity:gravity]) { return; }
+    printf("rawRotation before decay: %.2f\n", atan2(gravity.x, gravity.y) - M_PI);
```
Comment (kimjune01) [inline ios/RCTMotionManager.m]: should remove before merging

File: ios/RCTMotionManager.m [hunk 2]
```diff
@@ -68,42 +69,70 @@ - (BOOL) isFlatWithGravity:(CMAcceleration) gravity {
...
     double rawRotation = atan2(lastX, lastY) - M_PI;
-    double rotation = [strongSelf.class rotationWithLockState:_lockState rawRotation:rawRotation];
-    _initialRotationWhenUnlocking = rotation;
-    _rotationDeltaForUnlocking = rawRotation - rotation;
+
+//    double rawRotation = atan2(gravity.x, gravity.y) - M_PI;
+
+    printf("rawRotation after decay: %.2f\n", rawRotation);
+    if (rawRotation < -3.142) {
+      rawRotation = rawRotation + 6.283;
```
Comment (kimjune01) [inline ios/RCTMotionManager.m]: should be a separate variable, because it's no longer raw

Comment (kimjune01) [reply ios/RCTMotionManager.m]: i suggest normalizedRotation

Comment (kimjune01) [reply ios/RCTMotionManager.m]: also 6.283 should be M_PI * 2, and test for cases where rawRotation is less than -2PI

Comment (lichen19853) [reply ios/RCTMotionManager.m]: Naming suggestion sounds good. rawRotation will be within (-2*M_PI, 0]

**Pull request merged**